src/charm.py

```diff
@@ -134,15 +139,20 @@ def _on_update_status(self, event):
         self._check_status()
 
     def _check_status(self) -> bool:
-        """Check if we heve all needed components.
+        """Check if we have all needed components.
 
         - partition name
         - slurm installed
         - slurmctld available and working
         - munge key configured and working
         """
         if self._slurm_manager.needs_reboot:
-            self.unit.status = BlockedStatus("Machine needs reboot")
+            try:
+                self.unit.status = MaintenanceStatus("Rebooting...")
+                logger.debug("Scheduling machine reboot")
+                subprocess.run(["juju-reboot"], check=True)
```

Comment:
_interface_slurmd.py_ is meant for the slurmd integration iirc so I do not believe the `subprocess.run(...)` should be moved there. I think where it is now is fine since the `_check_status(...)` method is a part of the slurmd charm class. Ideally, the reboot mechanism would be moved to a "slurmd manager" (e.g. the need for a reboot is assessed after the slurm-wlm package is installed), but that is a more invasive refactor that should be discussed further.

Reply:
Ah yes, a "slurmd manager" is more in-line with what I was thinking and I agree we can discuss that further later on.